### PR TITLE
Fix subscribe and unsubscribe command and pattern mode

### PR DESCRIPTION
**Pull request by ShooterIT**
- Fix subscribe and psubscribe command reply, the return number
  is the sum of subscription and psubscription
- Fix unpsubscribe command operates on subscription data instead of
  psubscription
- The unsubscribe and punsubscribe commands can operates on more than
  one channel and pattern
- The unsubscribe and punsubscribe commands should have reply

### CODE REVIEW

File: src/redis_cmd.cc [hunk 1]
```diff
@@ -3270,57 +3270,72 @@ class CommandPublish : public Commander {
...
-  CommandUnSubscribe() : Commander("unsubcribe", -1, false) {}
+  CommandUnSubscribe() : Commander("unsubscribe", -1, false) {}
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    if (args_.size() > 1) {
-      conn->UnSubscribeChannel(args_[1]);
-    } else {
+    if (args_.size() == 1) {
       conn->UnSubscribeAll();
+      SubscribeCommmandReply(output, "unsubscribe", "",
+        conn->SubscriptionsCount() + conn->PSubscriptionsCount());
```
Comment (karelrooted) [inline src/redis_cmd.cc:3300]: in unsubscribe all, a message for every unsubscribed channel will be sent to the client in stock redis (verified on redis v5.0.4)， is this response on purpose？

File: src/redis_cmd.cc [hunk 2]
```diff
@@ -3270,57 +3270,72 @@ class CommandPublish : public Commander {
...
  public:
-  CommandPUnSubscribe() : Commander("punsubcribe", -1, false) {}
+  CommandPUnSubscribe() : Commander("punsubscribe", -1, false) {}
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    if (args_.size() > 1) {
-      conn->PUnSubscribeChannel(args_[1]);
+    if (args_.size() == 1) {
+      conn->PUnSubscribeAll();;
+      SubscribeCommmandReply(output, "punsubscribe", "",
+        conn->SubscriptionsCount() + conn->PSubscriptionsCount());
```
Comment (karelrooted) [inline src/redis_cmd.cc:3332]: in punsubscribe all, a message for every punsubscribed channel_patten will be sent to the client in stock redis (verified on redis v5.0.4)， is this response on purpose？

**Review by karelrooted — CHANGES_REQUESTED** [review]

File: src/redis_cmd.cc (hunk 2 above)
Comment (ShooterIT) [reply src/redis_cmd.cc]: You are right, i miss that, please have a look again.

**Review by karelrooted — APPROVED** [review]
LGTM

**Pull request merged**